src/rhsm/config.py

```diff
@@ -120,8 +120,9 @@ def __init__(self, config_file=None, defaults=None):
 
     def save(self, config_file=None):
         """Writes config file to storage."""
-        fo = open(self.config_file, "w")
-        self.write(fo)
+        with open(self.config_file + ".tmp", "w") as fo:
```

Comment:
This is better than previous solution, but I would like to improve it a little bit more. Why? Let's consider following situation. Two processes try to write to the `/etc/rhsm/rhsm.conf.tmp` in the same time. The configuration file can be corrupted, because self.write(fo) can be interrupted. I propose something like this:

```python
import tempfile
import os

dir_name = os.path.dirname(self.config_file)
temp_filename = tempfile.mktemp(prefix='rhsm.', suffix='.tmp', dir=dir_name)
with open(temp_filename, "w") as fo:
    self.write(fo)
os.rename(self.config_file + ".tmp", self.config_file)
```

As you can see each process creates unique temporary file and renaming should be atomic.

Reply:
Yes, I was thinking the same.

`mktemp` is deprecated, though.  I'll use `NamedTemporaryFile`.

Reply:
> `mktemp` is deprecated, though. I'll use `NamedTemporaryFile`.

Sure, the code was just suggestion how to solve the issue. :-)

Reply:
Done.